src/components/common/form/date-input/DateInput.jsx

```diff
@@ -164,15 +173,7 @@ const DateInputItem = ({
           data-testid={`${name}-${dateType}-input`}
           defaultValue={defaultValue}
           {...register(name + '-' + dateType, {
-            onChange: () => setDates(),
-            required: {
-              value: true,
-              message: 'Enter a ' + dateType,
-            },
-            pattern: {
-              value: pattern,
-              message: 'Enter a valid ' + dateType,
-            },
+            onChange: () => triggerOnChangeUpdated(),
```

Comment:
How come this logic has been moved into `triggerOnChangeUpdated` instead of just calling `setDates`?

Reply:
Updated functions to remove unnecessary steps